datasets/colours.py

```diff
@@ -31,10 +31,21 @@
 ]
 
 
+def match_color(text, colors):
+    """
+    Check if any of the exact words in 'colors' is in the text.
+    Returns 1 if any color is found as a whole word, otherwise 0.
+    """
+    colors_pattern = '|'.join([re.escape(color) for color in colors])
+    pattern = rf'(?<!\S)(?:{colors_pattern})(?!\S)|(?<!\S)(?:{colors_pattern})(?=[,.!?])'
```

Comment:
Is this `(?!\S)` an optional white space? I think you can write it just `\s?`, but I'm not sure.

Reply:
The rationale was the pattern (?<!\S) is a negative lookbehind for a non-whitespace character, which is more precise than \s? in this context. Although I did stop when tests were passing for the element so could be improved.

Reply:
I don't understand the difference 😅 But as you said, the important part is that it passes the tests. An important detail would then be to have negative examples too, so that we know that the evaluation is not too loose.